internal/uvm/vsmb.go

```diff
@@ -194,9 +205,32 @@ func (uvm *UtilityVM) GetVSMBUvmPath(ctx context.Context, hostPath string) (stri
 		hostPath, f = filepath.Split(hostPath)
 	}
 	hostPath = filepath.Clean(hostPath)
-	share, err := uvm.findVSMBShare(ctx, m, hostPath)
+	shareKey, err := getVSMBShareKey(hostPath, readOnly)
+	if err != nil {
+		return "", err
+	}
+	share, err := uvm.findVSMBShare(ctx, m, shareKey)
 	if err != nil {
 		return "", err
 	}
 	return filepath.Join(share.guestPath, f), nil
 }
+
+// getVSMBShareKey returns a string key which encapsulates the information that
+// is used to look up an existing VSMB share. If a share is being added, but
+// there is an existing share with the same key, the existing share will be used
+// instead (and its ref count incremented).
+func getVSMBShareKey(hostPath string, readOnly bool) (string, error) {
+	keyStruct := struct {
+		HostPath string
+		ReadOnly bool
+	}{
+		HostPath: hostPath,
+		ReadOnly: readOnly,
+	}
+	keyBytes, err := json.Marshal(keyStruct)
```

Comment:
Do we really need to marshal this as a json string? Can we just make a string by doing something like `fmt.Sprintf("%s-%s", HostPath, ReadOnly)`? We won't need this keyStruct in that case.

Reply:
It would be %t for readOnly but +1 on this suggestion

Reply:
Yeah, I can change it for now. There might be a point where struct marshaling is easier to maintain, but for only two fields using `Sprintf` seems good.